src/vsphere_cpi/lib/cloud/vsphere/resources/resource_pool.rb

```diff
@@ -27,12 +28,24 @@ def mob
         if @cluster_config.resource_pool.nil?
           @mob = @root_resource_pool
         else
-          client = @client
-          @mob = client.cloud_searcher.get_managed_object(
-            Vim::ResourcePool,
-            :root => @root_resource_pool,
-            :name => @cluster_config.resource_pool)
-          logger.debug("Found requested resource pool: #{@mob}")
+          begin
+            resource_pool_path_suffix = @cluster_config.resource_pool
+            datacenter_cluster_path_prefix = "#{@datacenter_name}/host/#{@cluster_config.name}/Resources/"
```

Comment:
This is dangerous if there are special characters in `@datacenter_name` or `@cluster_config.name`. I believe that you can make this safe by URL escaping those strings.

Reply:
Good Catch. thanks!